lib/src/directory_watcher/windows.dart

```diff
@@ -372,10 +372,26 @@ class _WindowsDirectoryWatcher
 
   /// Start or restart the underlying [Directory.watch] stream.
   void _startWatch() {
-    // Batch the events together so that we can dedup events.
-    var innerStream = Directory(path).watch(recursive: true);
-    _watchSubscription = innerStream.listen(_onEvent,
-        onError: _eventsController.addError, onDone: _onDone);
+    startInnerStream() {
+      // Batch the events together so that we can dedup events.
+      var innerStream = Directory(path).watch(recursive: true);
+      _watchSubscription = innerStream.listen(_onEvent,
+          onError: _eventsController.addError, onDone: _onDone);
+    }
+
+    // Start a zone to catch "Directory watcher closed unexpectedly."
+    runZoned(() {
+      startInnerStream();
```

Comment:
Can we skip the extra layer of function literal?

```dart
runZoned(startInnerStream,
    onError: (error, StackTrace stackTrace) {
```

Reply:
done